src/routes/notifications/notifications.controller.spec.ts

```diff
@@ -0,0 +1,243 @@
+  TestConfigurationModule,
+} from '../../config/__tests__/test.configuration.module';
+import { TestCacheModule } from '../../datasources/cache/__tests__/test.cache.module';
+import { NetworkResponseError } from '../../datasources/network/entities/network.error.entity';
+import {
+  mockNetworkService,
+  TestNetworkModule,
+} from '../../datasources/network/__tests__/test.network.module';
+import { DomainModule } from '../../domain.module';
+import { Chain } from '../../domain/chains/entities/chain.entity';
+import { chainBuilder } from '../../domain/chains/entities/__tests__/chain.builder';
+import { DataSourceErrorFilter } from '../common/filters/data-source-error.filter';
+import { SafeRegistration } from './entities/safe-registration.entity';
+import { registerDeviceDtoBuilder } from './entities/__tests__/register-device.dto.builder';
+import { safeRegistrationBuilder } from './entities/__tests__/safe-registration.builder';
+import { NotificationsModule } from './notifications.module';
+
+describe('Notifications Controller (Unit)', () => {
+  let app: INestApplication;
+
+  const safeConfigUrl = faker.internet.url();
+
+  beforeAll(async () => {
+    fakeConfigurationService.set('safeConfig.baseUri', safeConfigUrl);
+    fakeConfigurationService.set('exchange.baseUri', faker.internet.url());
+    fakeConfigurationService.set(
+      'exchange.apiKey',
+      faker.random.alphaNumeric(),
+    );
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        NotificationsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
+  });
+
+  describe('POST /register/notifications', () => {
+    it('Success', async () => {
+      const chains: Chain[] = range(4).map((idx) =>
+        chainBuilder().with('chainId', idx.toString()).build(),
+      );
+      const safeRegistrations: SafeRegistration[] = chains.map((chain) =>
+        safeRegistrationBuilder().with('chain_id', chain.chainId).build(),
+      );
+      const registerDeviceDto = registerDeviceDtoBuilder()
+        .with('safeRegistration', safeRegistrations)
+        .build();
+      mockNetworkService.get.mockImplementation((url) => {
+        const getChainUrlPattern = `${safeConfigUrl}/api/v1/chains/`;
+        if (url.includes(getChainUrlPattern)) {
+          return Promise.resolve({ data: sample(chains) });
+        }
+        return Promise.reject(`No matching rule for url: ${url}`);
+      });
+      mockNetworkService.post.mockImplementation(() => Promise.resolve());
+
+      await request(app.getHttpServer())
+        .post('/register/notifications')
+        .send(registerDeviceDto)
+        .expect(200);
```

Comment:
Should the body be empty?

Reply:
According to the Rust implementation, it should be empty when the device registration is successful. But when errors from the provider are caught, it should be indicated in the body response, I will change it 👍

Reply:
> According to the Rust implementation, it should be empty when the device registration is successful. But when errors from the provider are caught, it should be indicated in the body response, I will change it 👍

Not sure if I understood: what will you change given that the status code is 200?

My suggestion was more leaning towards validating the body in the response.

Reply:
Sorry I explained it poorly, I was meant I'm going to change it in the case of errors. In this case the body should be empty so it's OK I guess.

Reply:
> Sorry I explained it poorly, I was meant I'm going to change it in the case of errors. In this case the body should be empty so it's OK I guess.

If the implementation returns a body (which is not expected) then this would still pass right?

Reply:
I changed the approach in https://github.com/5afe/safe-client-gateway-nest/pull/267/commits/0728521aa558fa73c90498d9c13798e313936c93 

Now the test checks the body is empty in this scenario.